crossbeam-utils/src/atomic/atomic_cell.rs

```diff
@@ -292,6 +306,22 @@ impl<T: Copy + Eq> AtomicCell<T> {
     }
 }
 
+// `MaybeUninit` prevents `T` from being dropped, so we need to implement `Drop`
+// for `AtomicCell` to avoid leaks of non-`Copy` types.
+impl<T> Drop for AtomicCell<T> {
+    fn drop(&mut self) {
+        if mem::needs_drop::<T>() {
+            // SAFETY:
+            // - the mutable reference guarantees that no other threads are concurrently accessing the atomic data
+            // - the raw pointer passed in is valid because we got it from a reference
+            // - `MaybeUninit` prevents double dropping `T`
+            unsafe {
+                drop(self.as_ptr().read());
```

Comment:
I don't think this should copy the data before dropping. It could be quite big.

```suggestion
                ptr::drop_in_place(self.as_ptr());
```

Reply:
Changed to use drop_in_place. Thanks for the suggestion!